src/oops/SOLID/singleResponsibilityPrinciple/before/EmployeeRepository.java

```diff
@@ -16,4 +20,11 @@
 
         return Arrays.asList(anna, billy, steve, magda);
     }
+
+    public void save(Employee employee) throws IOException {
+        String employeeDetails = EmployeeSerializer.serializeEmployee(employee);
+        Path path = Paths.get(employee.getFullName()
+                .replace(" ","_") + ".rec");
+        Files.write(path, employeeDetails.getBytes());
```

Comment:
Persisting data could be moved out as a separately responsibility. The same could be used to persist other objects later. what do you think?

Reply:
My understanding was that if the persistence method changes like db instead of files, we can change the Repository class itself, but moving this responsibility  out makes more sense. I am not sure about using the same layer to persist other objects though. Persistence would be tied to object of a particular type, right?